lib/core/include/scipp/core/element/math.h

```diff
@@ -161,6 +161,10 @@ constexpr auto midpoint = overloaded{
           mid[i] = std::midpoint(a[i], b[i]);
         }
         return mid;
+      } else if constexpr (std::is_same_v<std::decay_t<decltype(a)>, int32_t>) {
+        return std::midpoint(static_cast<float>(a), static_cast<float>(b));
```

Comment:
Why not double? Float seems dangerous.

Reply:
I don't think it is the same issue. Precision loss with 64 bit is much much less likely.